### DISCUSSION TITLE
onModelConnected - iTwin Developer Program Video, Chap 4: Lesson 2 - Hiding Categories

### DISCUSSION THREAD

**Discussion by markstefanchuk** [iTwin/itwinjs-core#2453] [Q&A] [answered]
Looking for help with a change in the tutorial. The App.tsx file downloaded with the template is different than the one used in the tutorial (video and guide). I am attempting to add onIModelConnected and I am not sure where to assign it.

The original return function in App.tsx looks like this,

```ts
return (
    <div className="viewer-container">
      <Header
        handleLogin={onLoginClick}
        loggedIn={isAuthorized}
        handleLogout={onLogoutClick}
      />
      {isLoggingIn ? (
        <span>"Logging in...."</span>
      ) : (
        isAuthorized && (
          <Viewer
            contextId={process.env.IMJS_CONTEXT_ID ?? ''}
            iModelId={process.env.IMJS_IMODEL_ID ?? ''}
            authConfig={{ oidcClient: AuthorizationClient.oidcClient }}
          />
        )
      )}
    </div>
  );
```

The video shows that onIModelConnected={onIModelConnected} is to be added after the authConfig assignment. But I don't know where to put this in this newer version of the return.

The version included with the template looks like this,

```ts
  return (
    <ViewerContext.Provider
      value={{ contextId: currContextId, iModelId: currIModelId, authOptions }}
    >
      <div className="viewer-container">
        <ViewerStartup />
      </div>
    </ViewerContext.Provider>
  );
```

I tried to add it to ViewerStartup, but that doesn't appear to be correct. 

Suggestions welcome.

**Comment by aruniverse** [comment, marked as answer]
Good catch! The template was recently changed, and looks like the Accreditation Program tutorials need to be updated.

You should be able to add the `onImodelConnected` callback to the `<Viewer />` component as before, but that component has been moved to be inside the [ViewerStartup.tsx](https://github.com/iTwin/viewer/blob/master/packages/modules/cra-template-web-viewer/template/src/ViewerStartup.tsx).

Sorry for the inconvenience, we'll work on updating the tutorials as soon as possible.

**Reply by markstefanchuk** [reply]
Thanks. That worked.
